Index: sidplayfp/c64/Banks/ExtraSidBank.h
===================================================================
--- sidplayfp/c64/Banks/ExtraSidBank.h	(revision 1244)
+++ sidplayfp/c64/Banks/ExtraSidBank.h	(working copy)
@@ -50,6 +50,7 @@
     static unsigned int mapperIndex(int address) { return address >> 5 & (MAPPER_SIZE - 1); }
 
 public:
+    ExtraSidBank() : sid(0){}
     void reset()
     {
         if (sid)
