diff --git a/src/gui/kernel/qkeymapper_win.cpp b/src/gui/kernel/qkeymapper_win.cpp
index b88186e..1c28464 100644
--- a/src/gui/kernel/qkeymapper_win.cpp
+++ b/src/gui/kernel/qkeymapper_win.cpp
@@ -774,6 +774,9 @@ QList<int> QKeyMapperPrivate::possibleKeys(QKeyEvent *e)
 {
     QList<int> result;
 
+    if(e->nativeVirtualKey() > 255)
+        return result;
+
     KeyboardLayoutItem *kbItem = keyLayout[e->nativeVirtualKey()];
     if(!kbItem)
         return result;
