diff -U 3 -r librcc-0.2.12-old/CMakeLists.txt librcc-0.2.12-new/CMakeLists.txt
--- librcc-0.2.12-old/CMakeLists.txt	2010-07-16 07:56:52 +0000
+++ librcc-0.2.12-new/CMakeLists.txt	2014-03-19 08:41:10 +0000
@@ -4,12 +4,14 @@
 set(CMAKE_VERBOSE_MAKEFILE YES)
 set(CMAKE_BUILD_TYPE RELEASE)
 
+
+
 if (WIN32)
     SET(USR_FOLDER "C:/DEVEL/UNIX" CACHE PATH "Path to libraries and includes")
     SET(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES};${USR_FOLDER}/include)
     INCLUDE_DIRECTORIES(${USR_FOLDER}/include)
     
-    find_library(ZLIB_LIBRARY NAMES zlib PATHS ${USR_FOLDER} PATH_SUFFIXES "lib")
+    find_library(ZLIB_LIBRARY NAMES zlib z PATHS ${ZLIB_FOLDER} PATH_SUFFIXES "lib")
     if (NOT ZLIB_LIBRARY)
 	MESSAGE(FATAL_ERROR "Could not find zlib")
     endif()
@@ -19,7 +21,7 @@
 	MESSAGE(FATAL_ERROR "Could not find LibXML2")
     endif()
 
-    find_library(ICONV_LIBRARY NAMES libiconv PATHS ${USR_FOLDER} PATH_SUFFIXES "lib")
+    find_library(ICONV_LIBRARY NAMES libiconv PATHS ${ICONV_FOLDER} PATH_SUFFIXES "lib")
     if (NOT ICONV_LIBRARY)
 	MESSAGE(FATAL_ERROR "Could not find iconv")
     endif ()
@@ -34,7 +36,7 @@
 	SET(HAVE_RCD TRUE)
     endif()
 
-    find_library(WSOCK_LIBRARY NAMES wsock32)
+    find_library(WSOCK_LIBRARY NAMES wsock32 PATHS ${WSOCK_FOLDER} PATH_SUFFIXES "lib")
     if (NOT WSOCK_LIBRARY)
 	MESSAGE(FATAL_ERROR "Could not find wsock library")
     endif()
@@ -55,7 +57,7 @@
 CHECK_INCLUDE_FILES (sys/wait.h HAVE_SYS_WAIT_H)
 CHECK_INCLUDE_FILES (signal.h HAVE_SIGNAL_H)
 CHECK_INCLUDE_FILES (sys/un.h HAVE_SYS_UN_H)
-CHECK_INCLUDE_FILES (unistd.h HAVE_UNISTD_H)
+#CHECK_INCLUDE_FILES (unistd.h HAVE_UNISTD_H)
 CHECK_INCLUDE_FILES (pwd.h HAVE_PWD_H)
 CHECK_INCLUDE_FILES (dlfcn.h HAVE_DLFCN_H)
 CHECK_INCLUDE_FILES (fcntl.h HAVE_FCNTL_H)
diff -U 3 -r librcc-0.2.12-old/src/CMakeLists.txt librcc-0.2.12-new/src/CMakeLists.txt
--- librcc-0.2.12-old/src/CMakeLists.txt	2010-07-16 04:37:32 +0000
+++ librcc-0.2.12-new/src/CMakeLists.txt	2014-03-19 08:45:22 +0000
@@ -66,5 +66,5 @@
     ${HEADERS}
 )
 
-ADD_LIBRARY(rcc ${rcc_SRCS})
+ADD_LIBRARY(rcc SHARED ${rcc_SRCS})
 TARGET_LINK_LIBRARIES(rcc ${LIBXML2_LIBRARY} ${ICONV_LIBRARY} ${ENCA_LIBRARY} ${LIBRCD_LIBRARY} ${ZLIB_LIBRARY} ${WSOCK_LIBRARY})
diff -U 3 -r librcc-0.2.12-old/src/librcc.c librcc-0.2.12-new/src/librcc.c
--- librcc-0.2.12-old/src/librcc.c	2013-04-23 23:25:31 +0000
+++ librcc-0.2.12-new/src/librcc.c	2014-03-19 07:18:47 +0000
@@ -347,10 +347,10 @@
     if (!dbname) return -1;
 
     sprintf(dbname,"%s/.rcc/",rcc_home_dir);
-    mkdir(dbname, 00755);
+    mkdir(dbname);
     
     sprintf(dbname,"%s/.rcc/%s.db/",rcc_home_dir,name);
-    mkdir(dbname, 00755);
+    mkdir(dbname);
 
     ctx->db4ctx = rccDb4CreateContext(dbname, flags);
     free(dbname);	
diff -U 3 -r librcc-0.2.12-old/src/rccexternal.c librcc-0.2.12-new/src/rccexternal.c
--- librcc-0.2.12-old/src/rccexternal.c	2013-11-13 20:42:04 +0000
+++ librcc-0.2.12-new/src/rccexternal.c	2014-03-19 07:35:40 +0000
@@ -25,6 +25,7 @@
 
 #include "../config.h"
 
+
 #ifdef HAVE_UNISTD_H
 # include <unistd.h>
 #endif /* HAVE_UNISTD_H */
@@ -69,11 +70,23 @@
 static char *addr = NULL;
 #endif /* HAVE_UNISTD_H */
 
+
+#ifndef S_IXOTH
+#define S_IXOTH 0
+#endif
+
+#ifndef WNOHANG 
+#define WNOHANG 1 
+#endif 
+
 int rccExternalInit() {
 #ifdef HAVE_UNISTD_H
+
+#ifndef __MINGW32__
 # ifdef HAVE_SIGNAL_H
     struct sigaction act;
 # endif /* HAVE_SIGNAL_H */
+#endif // __MINGW32__
 # ifdef HAVE_SYS_STAT_H
     struct stat st;
 # endif /* HAVE_SYS_STAT_H */
@@ -90,12 +103,14 @@
 	if (pid == (pid_t)-1) return -1;
 	sprintf(addr,"%s/.rcc/comm/%lu.sock", rcc_home_dir, (unsigned long)pid);
 
+#ifndef __MINGW32__
 # ifdef HAVE_SIGNAL_H
 	act.sa_handler = SIG_IGN;
 	sigemptyset(&act.sa_mask);
 	act.sa_flags = 0;
 	sigaction(SIGPIPE,&act,NULL);
 # endif /* HAVE_SIGNAL_H */
+#endif //__MINGW32__
 
 	return 0;
     }
diff -U 3 -r librcc-0.2.12-old/src/rcclock.c librcc-0.2.12-new/src/rcclock.c
--- librcc-0.2.12-old/src/rcclock.c	2012-11-11 05:00:10 +0000
+++ librcc-0.2.12-new/src/rcclock.c	2014-03-19 07:36:39 +0000
@@ -63,10 +63,10 @@
     if (!stmp) return -1;
 
     sprintf(stmp,"%s/.rcc/", rcc_home_dir);
-    mkdir(stmp, 00755);
+    mkdir(stmp);
     
     sprintf(stmp,"%s/.rcc/locks/", rcc_home_dir);
-    mkdir(stmp, 00700);
+    mkdir(stmp);
 
     sprintf(stmp,"%s/.rcc/locks/rcc.lock", rcc_home_dir);
 
diff -U 3 -r librcc-0.2.12-old/src/rccxml.c librcc-0.2.12-new/src/rccxml.c
--- librcc-0.2.12-old/src/rccxml.c	2013-11-13 18:23:45 +0000
+++ librcc-0.2.12-new/src/rccxml.c	2014-03-19 08:14:55 +0000
@@ -340,7 +340,7 @@
     if (!config) return -1;
 
     sprintf(config,"%s/.rcc/",rcc_home_dir);
-    mkdir(config, 00755);
+    mkdir(config);
     
     sprintf(config,"%s/.rcc/%s.xml",rcc_home_dir,name);
     fd = open(config, O_CREAT|O_RDWR,00644);
