Index: Spc_Cpu.cpp
===================================================================
--- gme/Spc_Cpu.cpp	(revision 45)
+++ gme/Spc_Cpu.cpp	(revision 46)
@@ -433,9 +433,7 @@
 			#endif
 			
 			// Registers other than $F2 and $F4-$F7
-			//if ( reg != 2 && reg != 4 && reg != 5 && reg != 6 && reg != 7 )
-			// TODO: this is a bit on the fragile side
-			if ( ((~0x2F00 << (bits_in_int - 16)) << reg) < 0 ) // 36%
+			if ( reg != 2 && (reg < 4 || reg > 7) ) // 36%
 				cpu_write_smp_reg( data, time, reg );
 		}
 		// High mem/address wrap-around
