diff -U 3 -r librcc-0.2.12-old/CMakeLists.txt librcc-0.2.12-new/CMakeLists.txt
--- librcc-0.2.12-old/CMakeLists.txt	2010-07-16 07:56:52 +0000
+++ librcc-0.2.12-new/CMakeLists.txt	2014-05-29 08:27:05 +0000
@@ -8,8 +8,10 @@
     SET(USR_FOLDER "C:/DEVEL/UNIX" CACHE PATH "Path to libraries and includes")
     SET(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES};${USR_FOLDER}/include)
     INCLUDE_DIRECTORIES(${USR_FOLDER}/include)
+    INCLUDE_DIRECTORIES(${CMAKE_INSTALL_PREFIX}/include)
+    INCLUDE_DIRECTORIES(${CMAKE_INSTALL_PREFIX}/include/libxml2)
     
-    find_library(ZLIB_LIBRARY NAMES zlib PATHS ${USR_FOLDER} PATH_SUFFIXES "lib")
+    find_library(ZLIB_LIBRARY NAMES zlib z PATHS ${USR_FOLDER} PATH_SUFFIXES "lib")
     if (NOT ZLIB_LIBRARY)
 	MESSAGE(FATAL_ERROR "Could not find zlib")
     endif()
@@ -24,17 +26,17 @@
 	MESSAGE(FATAL_ERROR "Could not find iconv")
     endif ()
     
-    find_library(ENCA_LIBRARY NAMES libenca PATHS ${USR_FOLDER} PATH_SUFFIXES "lib")
+    find_library(ENCA_LIBRARY NAMES libenca PATHS ${USR_FOLDER} ${CMAKE_INSTALL_PREFIX} PATH_SUFFIXES "lib")
     if (ENCA_LIBRARY)
 	SET(HAVE_ENCA TRUE)
     endif()
     
-    find_library(LIBRCD_LIBRARY NAMES rcd PATHS ${USR_FOLDER} PATH_SUFFIXES "lib")
+    find_library(LIBRCD_LIBRARY NAMES rcd PATHS ${USR_FOLDER} ${CMAKE_INSTALL_PREFIX} PATH_SUFFIXES "lib")
     if (LIBRCD_LIBRARY)
 	SET(HAVE_RCD TRUE)
     endif()
 
-    find_library(WSOCK_LIBRARY NAMES wsock32)
+    find_library(WSOCK_LIBRARY NAMES wsock32 PATHS ${USR_FOLDER} PATH_SUFFIXES "lib")
     if (NOT WSOCK_LIBRARY)
 	MESSAGE(FATAL_ERROR "Could not find wsock library")
     endif()
diff -U 3 -r librcc-0.2.12-old/src/CMakeLists.txt librcc-0.2.12-new/src/CMakeLists.txt
--- librcc-0.2.12-old/src/CMakeLists.txt	2010-07-16 04:37:32 +0000
+++ librcc-0.2.12-new/src/CMakeLists.txt	2014-05-29 08:27:05 +0000
@@ -66,5 +66,5 @@
     ${HEADERS}
 )
 
-ADD_LIBRARY(rcc ${rcc_SRCS})
+ADD_LIBRARY(rcc SHARED ${rcc_SRCS})
 TARGET_LINK_LIBRARIES(rcc ${LIBXML2_LIBRARY} ${ICONV_LIBRARY} ${ENCA_LIBRARY} ${LIBRCD_LIBRARY} ${ZLIB_LIBRARY} ${WSOCK_LIBRARY})
diff -U 3 -r librcc-0.2.12-old/src/librcc.c librcc-0.2.12-new/src/librcc.c
--- librcc-0.2.12-old/src/librcc.c	2013-04-23 23:25:31 +0000
+++ librcc-0.2.12-new/src/librcc.c	2014-05-29 08:27:05 +0000
@@ -347,10 +347,18 @@
     if (!dbname) return -1;
 
     sprintf(dbname,"%s/.rcc/",rcc_home_dir);
+#ifdef _WIN32
+    mkdir(dbname);
+#else
     mkdir(dbname, 00755);
+#endif
     
     sprintf(dbname,"%s/.rcc/%s.db/",rcc_home_dir,name);
+#ifdef _WIN32
+    mkdir(dbname);
+#else
     mkdir(dbname, 00755);
+#endif
 
     ctx->db4ctx = rccDb4CreateContext(dbname, flags);
     free(dbname);	
diff -U 3 -r librcc-0.2.12-old/src/rccconfig.c librcc-0.2.12-new/src/rccconfig.c
--- librcc-0.2.12-old/src/rccconfig.c	2012-11-11 05:00:10 +0000
+++ librcc-0.2.12-new/src/rccconfig.c	2014-05-29 08:27:05 +0000
@@ -35,6 +35,8 @@
 rcc_language_alias rcc_default_aliases_embeded[RCC_MAX_ALIASES + 1] = {
 /*    { "cs_SK", "sk" },
     { "ru_UA", "uk" },*/
+    { "Russian_Russia", "ru" },
+    { "Ukrainian_Ukraine", "uk" },
     { NULL, NULL }
 };
 
diff -U 3 -r librcc-0.2.12-old/src/rccexternal.c librcc-0.2.12-new/src/rccexternal.c
--- librcc-0.2.12-old/src/rccexternal.c	2013-11-13 20:42:04 +0000
+++ librcc-0.2.12-new/src/rccexternal.c	2014-05-29 08:27:05 +0000
@@ -70,6 +70,7 @@
 #endif /* HAVE_UNISTD_H */
 
 int rccExternalInit() {
+#ifndef _WIN32
 #ifdef HAVE_UNISTD_H
 # ifdef HAVE_SIGNAL_H
     struct sigaction act;
@@ -111,10 +112,12 @@
     }
 # endif /* HAVE_SYS_STAT_H */
 #endif /* HAVE_UNISTD_H */
+#endif /* _WIN32 */
     _exit(1);
 }
 
 void rccExternalFree() {
+#ifndef _WIN32
 #ifdef HAVE_UNISTD_H
     int retry;
     pid_t res;
@@ -135,6 +138,7 @@
     pid = (pid_t)-1;
     if (addr) free(addr);
 #endif /* HAVE_UNISTD_H */
+#endif /* _WIN32 */
 }
 
 #ifdef HAVE_SYS_SELECT_H
diff -U 3 -r librcc-0.2.12-old/src/rcchome.c librcc-0.2.12-new/src/rcchome.c
--- librcc-0.2.12-old/src/rcchome.c	2012-11-11 05:00:10 +0000
+++ librcc-0.2.12-new/src/rcchome.c	2014-05-29 08:27:05 +0000
@@ -49,8 +49,11 @@
 #ifdef HAVE_PWD_H
     struct passwd *pw;
 #endif /* HAVE_PWD_H */
-
+#ifdef _WIN32
+    tmp = getenv ("HOMEPATH");
+#else
     tmp = getenv ("HOME");
+#endif
     if (tmp) rcc_home_dir = strdup (tmp);
 #ifdef HAVE_PWD_H
     else {
diff -U 3 -r librcc-0.2.12-old/src/rcclocale.c librcc-0.2.12-new/src/rcclocale.c
--- librcc-0.2.12-old/src/rcclocale.c	2013-04-22 23:06:05 +0000
+++ librcc-0.2.12-new/src/rcclocale.c	2014-05-29 08:27:05 +0000
@@ -90,7 +90,7 @@
 
     for (j=0;rcc_default_aliases[j].alias;j++) 
 	if (strncmp(l,rcc_default_aliases[j].alias,i)==0) {
-	    l = rcc_default_aliases[j].alias;
+	    l = rcc_default_aliases[j].lang;
 	    break;
 	}
 
diff -U 3 -r librcc-0.2.12-old/src/rcclock.c librcc-0.2.12-new/src/rcclock.c
--- librcc-0.2.12-old/src/rcclock.c	2012-11-11 05:00:10 +0000
+++ librcc-0.2.12-new/src/rcclock.c	2014-05-29 08:27:05 +0000
@@ -63,10 +63,18 @@
     if (!stmp) return -1;
 
     sprintf(stmp,"%s/.rcc/", rcc_home_dir);
+#ifdef _WIN32
+    mkdir(stmp);
+#else
     mkdir(stmp, 00755);
+#endif
     
     sprintf(stmp,"%s/.rcc/locks/", rcc_home_dir);
-    mkdir(stmp, 00700);
+#ifdef _WIN32
+    mkdir(stmp);
+#else
+    mkdir(stmp, 00755);
+#endif
 
     sprintf(stmp,"%s/.rcc/locks/rcc.lock", rcc_home_dir);
 
diff -U 3 -r librcc-0.2.12-old/src/rccxml.c librcc-0.2.12-new/src/rccxml.c
--- librcc-0.2.12-old/src/rccxml.c	2013-11-13 18:23:45 +0000
+++ librcc-0.2.12-new/src/rccxml.c	2014-05-29 08:27:05 +0000
@@ -340,7 +340,11 @@
     if (!config) return -1;
 
     sprintf(config,"%s/.rcc/",rcc_home_dir);
+#ifdef _WIN32
+    mkdir(config);
+#else
     mkdir(config, 00755);
+#endif
     
     sprintf(config,"%s/.rcc/%s.xml",rcc_home_dir,name);
     fd = open(config, O_CREAT|O_RDWR,00644);
