diff -U 3 -r librcd-0.1.14-old/src/Makefile.am librcd-0.1.14-new/src/Makefile.am
--- librcd-0.1.14-old/src/Makefile.am	2009-08-27 19:57:40 +0000
+++ librcd-0.1.14-new/src/Makefile.am	2014-03-18 10:46:42 +0000
@@ -2,5 +2,5 @@
 librcd_la_SOURCES = librcd.c russian_table.h
 include_HEADERS = librcd.h
 
-librcd_la_LDFLAGS = -version-info @LIBRCD_VERSION_INFO@
+librcd_la_LDFLAGS = -version-info @LIBRCD_VERSION_INFO@ -no-undefined
 
